content/concepts/did-ddo.md

```diff
@@ -1,173 +1,521 @@
 
-Each service has an `attributes` section holding the information related to the service. That section _must_ have a `main` sub-section, holding all the mandatory information that a service has to provide.
+#### Compute Privacy
 
-A part of the `attributes.main` sub-section, other optional sub-sections like `attributes.extra` can be added. These depend on the service type.
+An asset with a service of `type` `compute` has the following additional attributes under the `privacy` object. This object is required if the asset is of `type` `compute`, but can be omitted for `type` of `access`.
 
-Each service has a `timeout` (in seconds) section describing how long the sevice can be used after consumption is initiated. A timeout of 0 represents no time limit.
+| Attribute                                  | Type                                  | Required | Description                                                                                                            |
+| ------------------------------------------ | ------------------------------------- | -------- | ---------------------------------------------------------------------------------------------------------------------- |
+| **`allowRawAlgorithm`**                    | `boolean`                             | **✓**    | If `true`, a drag & drop algorithm can be run                                                                          |
+| **`allowNetworkAccess`**                   | `boolean`                             | **✓**    | If `true`, the algorithm job will have network access (still WIP)                                                      |
+| **`publisherTrustedAlgorithmPublishers `** | Array of `String`                     | **✓**    | If empty, then any published algorithm is allowed. Otherwise, only published algorithms by some publishers are allowed |
+| **`publisherTrustedAlgorithms `**          | Array of `publisherTrustedAlgorithms` | **✓**    | If empty, then any published algorithm is allowed. (see below)                                                         |
 
-The `cost` attribute is obsolete, as of Ocean V3. As of V3, to consume an asset, one sends exactly 1.0 datatokens of the asset, so a `cost` is not needed.
+The `publisherTrustedAlgorithms ` is an array of objects with the following structure:
 
-## DDO Service Example
+| Attribute                      | Type     | Required | Description                                                        |
+| ------------------------------ | -------- | -------- | ------------------------------------------------------------------ |
+| **`did`**                      | `string` | **✓**    | The DID of the algorithm which is trusted by the publisher.        |
+| **`filesChecksum`**            | `string` | **✓**    | Hash of algorithm's `encryptedFiles` + `files` section (as string) |
```

Comment:
this needs updating if we ditch huge `files` object. Something else needs to be checksummed here, like the encrypted `files` hash string